financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.extensions/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/extensions/authorize/impl/ExternalAPIConsentRetrievalStep.java

```diff
@@ -82,10 +82,17 @@ public void execute(ConsentData consentData, JSONObject jsonObject) throws Conse
             }
 
             ExternalAPIConsentResourceRequestDTO externalAPIConsentResource = null;
+            String consentFileContent = "";
             if (consentId != null) {
                 DetailedConsentResource detailedConsentResource = consentCoreService.getDetailedConsent(consentId);
                 if (detailedConsentResource != null) {
                     externalAPIConsentResource = new ExternalAPIConsentResourceRequestDTO(detailedConsentResource);
+                    try {
+                        consentFileContent = consentCoreService.getConsentFile(consentId).getConsentFile();
```

Comment:
will this line cause a NPE? Will consentCoreService.getConsentFile(consentId)==null for other consent types?

Reply:
Consent core service throws a ConsentManagementException if a file is not found for the given consent id. We are handling it here.